library/Icingadb/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -214,13 +214,11 @@ protected function fetchColumnSuggestions($searchTerm)
         $titleAdded = false;
         foreach ($this->getDb()->select($this->queryCustomvarConfig($searchTerm)) as $customVar) {
             $search = $name = $customVar->flatname;
-            if (preg_match('/\w+\[(\d+)]$/', $search, $matches)) {
-                // array vars need to be specifically handled
-                if ($matches[1] !== '0') {
-                    continue;
-                }
-
-                $name = substr($search, 0, -3);
+            if (preg_match('/\w+\[(\d*)]$/', $search, $matches)) {
+                // Count the digits of the used array index and add the len of the curly brackets (2). The hardcoded
+                // previously used value `-3` may fail for array indices `>= 10`.
+                $length = $matches[1] === "" ? 2 : (int) log10((int) $matches[1]) + 1 + 2;
```

Comment:
Isn't that just a complicated (and maybe even buggy) way of writing `$length = strlen($matches[1]) + 2;`

> The hardcoded previously used value `-3` may fail for array indices `>= 10`.

That sounds more like something for a commit message rather than an inline comment (there's no trace of -3 in the new code).

Reply:
> Isn't that just a complicated (and maybe even buggy) way of writing `$length = strlen($matches[1]) + 2;`

Don't see what makes these changes " buggy". Yes, it might be quite complex for such a trivial task, but after all it only performs a simple mathematical operation.

Reply:
```
php > var_dump((int) log10((int) "0"));
int(0)
php > var_dump(log10((int) "0"));
float(-INF)
```

I mean probably works, but only because of some PHP quirks.

You want the length of a string to remove it from some other string, why would you want to resort to some floating point number calculations when you can just use the length directly?